File: 141-linked-list-cycle/141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        auto slow = head,fast = head;
        if (!fast||!fast->next) return false;
        while(fast&&fast->next){
            fast = fast->next->next;
            if(fast==slow) return true;
            slow = slow->next;
        }
        
        return fast==slow;
    }
};

File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        slow,fast=  head,head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
            if fast==slow:
                break
        else:
            return False
        return True
        